tests/cephfs/cephfs_nfs/nfs_ha_create_using_spec_ingress.py

```diff
@@ -0,0 +1,149 @@
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    try:
+        fs_util = FsUtils(ceph_cluster)
+        config = kw.get("config")
+        clients = ceph_cluster.get_ceph_objects("client")
+        build = config.get("build", config.get("rhbuild"))
+        fs_util.prepare_clients(clients, build)
+        fs_util.auth_list(clients)
+        log.info("checking Pre-requisites")
+        client1 = clients[0]
+        # get nfs nodes
+        nfs_list = ceph_cluster.get_ceph_objects("nfs")
+        log.info(nfs_list)
+        virtual_ip = "10.8.128.100"
+        subnet = "21"
+        nfs_ID = "cephfs_nfs"
+        port = "12345"
+        data = {
+            "service_type": "nfs",
+            "service_id": nfs_ID,
+            "placement": {
+                "hosts": [nfs_list[0].node.hostname, nfs_list[1].node.hostname],
+                "count": 1,
+            },
+            "spec": {"port": port},
+        }
+        mounting_dir = "".join(
+            random.choice(string.ascii_lowercase + string.digits)
+            for _ in list(range(10))
+        )
+        nfs_mounting_dir = f"/mnt/cephfs_nfs{mounting_dir}_1/"
+        client1.exec_command(sudo=True, cmd=f"mkdir -p {nfs_mounting_dir}")
+        client1.exec_command(sudo=True, cmd=f"echo '{data}' > /tmp/nfs.yaml")
+        client1.exec_command(sudo=True, cmd="ceph orch apply -i /tmp/nfs.yaml")
+        # validate_service
+        fs_util.validate_service(client1, f"nfs.{nfs_ID}")
+        ingress = {
+            "service_type": "ingress",
+            "service_id": f"nfs.{nfs_ID}",
+            "placement": {
+                "hosts": [nfs_list[0].node.hostname, nfs_list[1].node.hostname],
+                "count": 1,
+            },
+            "spec": {
+                "backend_service": f"nfs.{nfs_ID}",
+                "frontend_port": "2049",
+                "monitor_port": "9000",
+                "virtual_ip": f"{virtual_ip}/{subnet}",
+            },
+        }
+        client1.exec_command(sudo=True, cmd=f"echo '{ingress}' > /tmp/ingress.yaml")
+        client1.exec_command(sudo=True, cmd="ceph orch apply -i /tmp/ingress.yaml")
+        # validate_service
+        fs_util.validate_service(client1, f"ingress.nfs.{nfs_ID}")
+        out2, ec2 = client1.exec_command(
+            sudo=True,
+            cmd=f"ceph orch ls --service_name=ingress.nfs.{nfs_ID} --format json-pretty",
+        )
+        output2 = json.loads(out2)
+        log.info(output2[0])
+        if f"ingress.nfs.{nfs_ID}" in output2[0]["service_name"]:
+            log.info("ingress service created successfully")
+        else:
+            raise CommandFailed("Failed to create ingress service")
+        if output2[0]["placement"]["count"] == 1:
+            log.info("ingress service created successfully")
+        else:
+            raise CommandFailed("Failed to create ingress service")
```

Comment:
Hi Julius,
Can you add the failure info also in exception message
"CommandFailed("Failed to create ingress service in validating count")"
As all the statements are same we might face difficulty when we hit an issue

rest looks good

Reply:
yeah actually I uploaded wrong logs...
updated the script with updated logs